pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -0,0 +1,343 @@
+// and what is in the CnsFileAccessConfig.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx, log := logger.GetNewContextWithLogger()
+	// Fetch the CnsFileAccessConfig instance
+	instance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Infof("CnsFileAccessConfig resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		log.Errorf("Error reading the CnsFileAccessConfig with name: %q on namespace: %q. Err: %+v",
+			request.Name, request.Namespace, err)
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Initialize backOffDuration for the instance, if required.
+	backOffDurationMapMutex.Lock()
+	var timeout time.Duration
+	if _, exists := backOffDuration[instance.Name]; !exists {
+		backOffDuration[instance.Name] = time.Second
+	}
+	timeout = backOffDuration[instance.Name]
+	backOffDurationMapMutex.Unlock()
+
+	// If the CnsFileAccessConfig instance is already successful,
+	// and not deleted by the user, remove the instance from the queue.
+	if instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Cleanup instance entry from backOffDuration map
+		backOffDurationMapMutex.Lock()
+		delete(backOffDuration, instance.Name)
+		backOffDurationMapMutex.Unlock()
+		return reconcile.Result{}, nil
+	}
+
+	log.Infof("Reconciling CnsFileAccessConfig with instance: %q from namespace: %q. timeout %q seconds", instance.Name, instance.Namespace, timeout)
+	if !instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Get the virtualmachine instance
+		vm, err := getVirtualMachine(ctx, instance.Spec.VMName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get virtualmachine instance for the VM with name: %q. Error: %+v", instance.Spec.VMName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Found virtualMachine instance for VM with name: %q: +%v", instance.Spec.VMName, vm)
+
+		// Set ownerRef on CnsFileAccessConfig instance (in-memory) to VM instance
+		setInstanceOwnerRef(instance, instance.Spec.VMName, vm.UID)
+
+		cnsFinalizerExists := false
+		// Check if finalizer already exists.
+		for _, finalizer := range instance.Finalizers {
+			if finalizer == cnsoperatortypes.CNSFinalizer {
+				cnsFinalizerExists = true
+				break
+			}
+		}
+		// Update finalizer and ownerRef on CnsFileAccessConfig instance
+		if !cnsFinalizerExists {
```

Comment:
This assumes that if CNSFinalizer exists on the instance then OwnerRef does too. Are we sure about this assumption?

Reply:
Looks like the comment is wrong. Updated